processor/deltatocumulativeprocessor/reorderprocessor/internal/gmetric/metric.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gmetric // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/reorderprocessor/internal/gmetric"
```

Comment:
@RichieSams I created yet another metrics package :(

I would eventually like `internal/[exp]/metrics` to become a generally useful, modern day (read probably generics) metrics api, fitting nicely next to `pmetric` but giving more type safety.

For the time being I am not sure we understand all requirements how such an api could look like however, so I decided to take another shot and the specific needs to for this code and it turned out quite differently than what I did for deltatocumulative before. I like this one slightly more. 

I'd suggest we keep building those packages separately for deltatocumulative, interval and reorder for now and try to come up with something generally useful once we have some more experience with them.

what do you think?

Reply:
That's fine by me. :)

I agree that we still don't fully know what interfaces / APIs we want. So we can just keep iterating for now. I'm adding some new functionality here that you may be interested in: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32794